### PR TITLE
Prematch now cleans up boosts

### PR DESCRIPTION
**Pull request by DBmaoha**
After game entering `eGameState.Prematch` multiple times, as for roundbased gamemodes, player's boosts will be cleaned. Same as vanilla does.

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Wait so in Northstar in roundbased modes, boosts carry over between rounds currently? o_O

**Comment by DBmaoha** [comment]
> Wait so in Northstar in roundbased modes, boosts carry over between rounds currently? o_O

yes they do, player can stack many hard covers or something for their next round, or in ctf carrying multiple turrets

**Comment by ASpoonPlaysGames** [comment]
> Wait so in Northstar in roundbased modes, boosts carry over between rounds currently? o_O

#397

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 1]
```diff
@@ -143,7 +152,8 @@ void function PlayerInventory_PopInventoryItem( entity player )
 
 void function PlayerInventory_RefreshEquippedState( entity player )
 {
-
+	player.TakeOffhandWeapon( OFFHAND_INVENTORY )
+	file.playerInventoryStacks[ player ] = []
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:156]: I'd prefer .clear here but whatever
```suggestion
	file.playerInventoryStacks[ player ].clear()
```

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 2]
```diff
@@ -19,13 +19,22 @@ void function Sv_ItemInventory_Init()
...
 	AddCallback_OnPlayerGetsNewPilotLoadout( Sv_ItemInventory_OnPlayerGetsNewPilotLoadout )
+	AddCallback_GameStateEnter( eGameState.Prematch, PrematchClearInventory )
 }
 
 void function Sv_ItemInventory_OnClientConnected( entity player )
 {
 	file.playerInventoryStacks[ player ] <- []
 }
 
+void function PrematchClearInventory()// vanilla behavior
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:30]:
```suggestion
void function PrematchClearInventory() // vanilla behavior
```

**Review by uniboi — APPROVED** [review]

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut (hunk 1 above)
Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut]: Wait that exists? Cool.

**Review by F1F7Y — APPROVED** [review]
Code looks good

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 3]
```diff
@@ -152,7 +161,7 @@ void function PlayerInventory_TakeAllInventoryItems( entity player )
 
 void function PlayerInventory_RefreshEquippedState( entity player )
 {
-
+
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:164]: Random whitespace added
```suggestion

```

Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:164]:
```suggestion

```

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 4]
```diff
@@ -152,7 +161,6 @@ void function PlayerInventory_TakeAllInventoryItems( entity player )
 
 void function PlayerInventory_RefreshEquippedState( entity player )
 {
-
 }
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:164]: oh my god github suggestion why did you delete the line instead of just the whitespace

File: Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut [hunk 5]
```diff
@@ -152,7 +161,6 @@ void function PlayerInventory_TakeAllInventoryItems( entity player )
 
 void function PlayerInventory_RefreshEquippedState( entity player )
 {
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut:163]:
```suggestion
{

```
kinda un-does a weird github suggestion moment from earlier, whoops

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks good, and works in testing
